winterfell/webapp/src/js/app.js

```diff
@@ -96,7 +96,22 @@ app.config(['$stateProvider', '$urlRouterProvider', function ($stateProvider, $u
     name: 'root.claimselect',
     url: '/claim/{claimId}/select-forms',
     templateUrl: 'templates/claimSelectForms.html',
-    controller: 'claimSelectFormsCtrl'
+    controller: 'claimSelectFormsCtrl',
+    resolve: {
+      forms: ['net', '$q', '$stateParams', function(net, $q, $stateParams) {
+        var deferred = $q.defer();
+
+        net.getFormsForClaim($stateParams.claimId).then(
```

Comment:
I like the resolve feature for two reasons:

- Child states inherit the resolved objects, which lets us avoid rewriting all the code to make API requests.
- Having the result of the resolve injected into the controller will make it easier to unit test the controllers when we eventually write frontend unit tests.